packages/sourcecred/src/core/credequate/config.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type {TimestampMs} from "../../util/timestamp";
+import type {WeightOperand} from "./contribution";
+import {type Operator, OPERATOR_KEY_PREFIX, OPERATORS} from "./operator";
+
+/**
+Semantically, allows weight configuration of different qualities/characteristics of
+contributions.
+
+Technically, a once-nested key-value store that maps key-subkey pairs to weights and
+specifies a default weight at the key-level that can be used when a queried subkey
+is not found.
+
+A Discord-based example might look like: 
+{
+  "key": "channel",
+  "default": 1,
+  "subkeys": [
+    { "subkey": "12345678", memo: "props", weight: 3 }
+  ]
+}
+ */
+export type WeightConfig = $ReadOnlyArray<{|
+  /** Arbitrary string enumerated by a plugin developer. */
+  +key: string,
+  /** The weight used when a queried subkey is not found. */
+  +default: number,
+  /** A key-value store mapping platform-based identifiers to weights. */
+  +subkeys: $ReadOnlyArray<{|
+    /** Potentially arbitrary, but most likely a platform-specific identifier */
+    +subkey: string,
+    /** An optional human-readable description of the subkey */
+    +memo?: string,
+    /** An algebraic constant/coefficient that defines the semantic value of
+    the subkey */
+    +weight: number,
+  |}>,
+|}>;
+/**
+A key-value store of configured operators, allowing the configuration of
+operators within an expression. For example, one might be able to
+configure that emoji reactions be added or multiplied.
+ */
+export type OperatorConfig = $ReadOnlyArray<{|
+  /** Arbitrary string enumerated by a plugin developer. NOT prefixed. */
+  +key: string,
+  /** The operator that should be applied wherever the key is found. */
+  +operator: Operator,
+|}>;
+/**
+A key-value store of how many shares different types of participation are worth.
+Shares are arbitrary numbers that compete in a zero-sum divvying of a
+ScoredContribution's score.
+ */
+export type SharesConfig = $ReadOnlyArray<{|
+  /** Arbitrary string enumerated by a plugin developer. */
+  +key: string,
+  /**
+  Number of shares allocated for the participation type indicated by the
+  key. Will be scored relative to the other share amounts within a contribution.
+  */
+  +amount: number,
+|}>;
+
+/**
+Wraps the other config types, and defines a time scope via a start timestamp.
+The end timestamp will be inferred as the next highest timestamp in an array
+of Configs.
+ */
+export type Config = {|
+  /** A note or a human-readable description to make it easier to recognize this config. **/
+  +memo: string,
+  +startTimeMs: TimestampMs,
+  +weights: WeightConfig,
+  +operators: OperatorConfig,
+  +shares: WeightConfig,
```

Comment:
@blueridger just double-checking if that `shares` field supposed to be of `SharesConfig` type?

Reply:
Good eye. I intentionally changed it to WeightBonfig, just forgot to remove the SharesConfig type since it is no longer used. Feel free to remove SharesConfig if you feel like doing a quick merge.

Reply:
Yeah seems already done in [e6c6147126e535d6149f49cd734715464f9784d8](https://github.com/sourcecred/sourcecred/commit/e6c6147126e535d6149f49cd734715464f9784d8)